GoogleForCreators/web-stories-wp
# Editor: remove mask from text element output

## Summary
- Per discussion [in the text sets channel](https://xwp.slack.com/archives/C01A6TD8HBJ/p1598445275005300) we should not apply `.mask`/`overflow: hidden` to all elements. This PR checks for the text type and removes the class which applied `overflow: hidden` to the text. Removing `.mask` from all elements



## Relevant Technical Choices

- removing `overflow: hidden` from all elements causes the video and image element crop to fail, so I **only** removed `.mask` from `text` type elements.



## To-do

- [x] percy screenshot tests
- [x] select a fancy font that bleeds over instead of basic `addTextElement`. Right now my tests are kind of lame... 
- [x] e2e test is more suitable for a Jest unit test in assets/src/edit-story/elements/text/test/output.js; less overhead. Could use a more generalized not.toHaveStyle('overflow: hidden') assertion instead of checking for the class name specifically.

I wasn't able to replicate the font cut-off from the ticket so steps to reproduce that might be useful...



## User-facing changes

- should fix the issue where certain text and glyphs would not overflow their containing box, resulting in undesired cropping e.g. 

![Screen Shot 2020-11-03 at 4 02 07 PM](https://user-images.githubusercontent.com/41136059/98050357-c20d3700-1dee-11eb-9b98-b924212a135a.png)
vs 
![Screen Shot 2020-11-03 at 4 01 29 PM](https://user-images.githubusercontent.com/41136059/98050367-c76a8180-1dee-11eb-8cef-71d69d7ff878.png)



## Testing Instructions
- create a new story
- add a text element to the page
- select "preview" 
- inspect the text element
- confirm its containing element no longer has the "mask" class:

### with mask 
![Screen Shot 2020-11-05 at 3 13 17 PM](https://user-images.githubusercontent.com/41136059/98302364-7aff7d00-1f79-11eb-94ab-2bda8b045a89.png)

### without mask
![Screen Shot 2020-11-05 at 3 08 06 PM](https://user-images.githubusercontent.com/41136059/98302393-8652a880-1f79-11eb-9446-5a9197aeacb4.png)




---



Fixes #4721
